src/odemis/model/_futures.py

```diff
@@ -489,7 +489,7 @@ def set_running_or_notify_cancel(self):
             # Set start to current time and end to estimated endtime (difference between current end and start)
             start, end = self.get_progress()
             startt = time.time()
-            endt = startt + end - start
+            endt = startt + (end - start)  # Force the computation order to reduce floating point error
```

Comment:
The brackets only add some clarity for the user (it doesn't change any mathematics) though that is fine.

Reply:
No. As the comment indicates, it does change the value. From a simple mathematics perspective, it shouldn't. However, this is float computation, not perfect mathematics numbers. So there are floating point errors that get introduced.
You can recreate the issue with something like this:
```python
startt = 1633339732.6324744
endt = startt + 1
newt = startt + 125.78
print(newt + endt - startt)
1633339859.4124742
print(newt + (endt - startt))
1633339859.4124744
```